build-system/tasks/release/index.js

```diff
@@ -128,10 +134,20 @@ function logSeparator_() {
  *
  * @param {string} outputDir full directory path to emplace artifacts in.
  * @param {string} tempDir full directory path to temporary working directory.
+ * @param {boolean} retainCustomConfig whether to exclude custom overlay and
+ * release configs from cleanup.
  * @return {Promise<void>}
  */
-async function prepareEnvironment_(outputDir, tempDir) {
-  execOrDie('amp clean');
+async function prepareEnvironment_(outputDir, tempDir, retainCustomConfig) {
+  let cleanCmd = 'amp clean';
+  if (retainCustomConfig) {
+    const cleanExclusions = [
+      CUSTOM_FLAVORS_CONFIG_PATH,
+      CUSTOM_OVERLAY_CONFIG_PATH,
+    ].join(',');
+    cleanCmd += ` --exclude ${cleanExclusions}`;
+  }
+  execOrDie(cleanCmd);
```

Comment:
Did you check whether `amp clean` removes the custom config? I don't believe `amp clean` removes files outside of `dist/`, `build/` etc... // cc: @rsimha

Reply:
Yes: https://github.com/ampproject/amphtml/blob/c8092fd78ae1b9c89450dcdbcaf71330dd206028/build-system/tasks/clean.js#L33

I would prefer that `build-system/global-configs/custom-config.json` and new config `build-system/global-configs/custom-flavor-configs.json` were not included in `amp clean`, but it's easy to argue both ways:
- These are not generated files; they are written by a user. So, in that sense, it makes sense to keep them around.
- It's easy to forget about them, and then you're left wondering why your builds don't behave like the AMP Project's builds.

Maybe a compromise would be to not delete by by default, but to emit a yellow warning that `amp clean` left them in place? This would simplify the current pull request..

Reply:
I like the idea of `amp clean` warning users about config-altering files that it doesn't delete. @rsimha wdyt?

Reply:
Just catching up with this thread. I couldn't have said it better than Matt: there are good reasons to clean the custom config file, and other good reasons to not clean it.

We already have a flag called `--include_subpackages` that optionally cleans all subpackage `node_modules` directories. Would it help to add a new flag `--include_custom_configs` that deletes any custom config files found locally? (This will address Matt's first point.)

And re: warning the developer when a custom config _is_ present, I think it should be printed by `amp build` and `amp dist` which are are more likely to be used than `amp clean`. WDYT?

Reply:
In commit 8bde827,
- `amp clean` emits: "Notice: Skipping build-system/global-configs/custom-config.json" (and similarly for `custom-flavor-configs.json`) if custom configs are detected
- `amp clean` supports new flag `--include_custom_configs` to also cleanup custom configs
- `amp dist`, `amp build`, `amp release` all emit notices similar to "Notice: prod config overlaid with custom-config.json" when a custom config overlay is applied
- `amp release` emits "Notice: release flavors supplemented by custom-flavors-config.json" when custom flavors are loaded